n2y/main.py

```diff
@@ -61,67 +68,83 @@ def name_column_valid(raw_rows, name_column):
+        logger.critical(
+            'Column "%s" does not contain a string.\n%s', name_column,
+            # only show columns that have strings as possible options
+            "Available column(s): " + ", ".join(available_columns()))
         return False
     return True
 
 
 def export_markdown(client, raw_rows, options):
     file_names = []
+    skips = {'empty': 0, 'unnamed': 0, 'duplicate': 0}
     for row in raw_rows:
         meta = simplify.flatten_database_row(row)
         page_name = meta[options.name_column]
-        filename = re.sub(r"[/,\\]", '_', page_name.lower())
         if page_name is not None:
-            print(f"Processing {meta[options.name_column]}", file=sys.stderr)
-            if filename in file_names:
-                print(f'WARNING: duplicate file name "{filename}.md"', file=sys.stderr)
-            file_names.append(filename)
-
-            pandoc_output = converter.load_block(client, row['id']).to_pandoc()
-            # do not create markdown pages if there is no page in Notion
-            if pandoc_output:
-                markdown = pandoc.write(pandoc_output, format='gfm+tex_math_dollars') \
-                    .replace('\r\n', '\n')  # Deal with Windows line endings
-
-                # create target path if it doesn't exist
-                # if not os.path.exists(options.target):
-                os.makedirs(options.target, exist_ok=True)
-
-                # sanitize file name just a bit
-                # maybe use python-slugify in the future?
-                with open(os.path.join(options.target, f"{filename}.md"), 'w') as f:
-                    f.write('---\n')
-                    f.write(yaml.dump(meta))
-                    f.write('---\n\n')
-                    f.write(markdown)
+            filename = re.sub(r"[\s/,\\]", '_', page_name.lower())
+            if filename not in file_names:
+                file_names.append(filename)
+
+                pandoc_output = converter.load_block(client, row['id']).to_pandoc()
+                # do not create markdown pages if there is no page in Notion
+                if pandoc_output:
+                    logger.info('Processing page "%s".', meta[options.name_column])
+                    markdown = pandoc.write(pandoc_output, format='gfm+tex_math_dollars') \
+                        .replace('\r\n', '\n')  # Deal with Windows line endings
+
+                    os.makedirs(options.target, exist_ok=True)
+
+                    # sanitize file name just a bit
+                    # maybe use python-slugify in the future?
+                    with open(os.path.join(options.target, f"{filename}.md"), 'w') as f:
+                        f.write('---\n')
+                        f.write(yaml.dump(meta))
+                        f.write('---\n\n')
+                        f.write(markdown)
+                else:
+                    logger.debug('Skipping page "%s" because it is empty.', page_name)
```

Comment:
I think it's appropriate that this message is at the "debug" level, since it's usually not interesting and is only useful for debugging a problem.